compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -384,16 +380,16 @@ fn compare_predicate_entailment<'tcx>(
                 &terr,
                 false,
             );
-            diag.emit();
-            return Err(ErrorReported);
+            let reported = diag.emit();
+            return Err(reported);
         }
 
         // Check that all obligations are satisfied by the implementation's
         // version.
         let errors = inh.fulfillment_cx.borrow_mut().select_all_or_error(&infcx);
         if !errors.is_empty() {
             infcx.report_fulfillment_errors(&errors, None, false);
-            return Err(ErrorReported);
+            return Err(ErrorReported::new_i_know_what_im_doing());
```

Comment:
`report_fulfillment_errors` should return `ErrorReported`.

Reply:
It seems that it just buffers the errors for later. I added a `delay_span_bug`.